library/std/src/sys/wasi/thread.rs

```diff
@@ -62,7 +162,17 @@ impl Thread {
     }
 
     pub fn join(self) {
-        self.0
+        cfg_if::cfg_if! {
+            if #[cfg(target_feature = "atomics")] {
+                unsafe {
+                    let ret = libc::pthread_join(self.id, ptr::null_mut());
```

Comment:
`join()` function is expected to return the value returned by the callback passed to `spawn()` - the second argument of the `pthread_join` is a pointer for that return value, but we pass `null` rather than retrieving the value. Should we try to get the value and return it from the `join()` on success?

Reply:
Signature for all platforms now looks like
`fn join(self)`
and it doesn't provide an opportunity to get a return value from it(without changing signature) rather than modifying `self`

@abrown @alexcrichton was it done on purpose? Can I make signature of `join` similar to `pthread_join` at least for `wasm32_wasi_threads` or it should stay consistent?

Reply:
For passing the thread return value we use a separate platform independent mechanism: https://github.com/rust-lang/rust/blob/fd0a3313f7a64cb16533030e49a271db449368c3/library/std/src/thread/mod.rs#L1429